### PR TITLE
feat: support checkout tidb-test with pr in batch merge

### PR DESCRIPTION
**Pull request by purelind**
When checking out the tidb-test code, it is necessary to determine whether the tidb-test PR has been specified based on the tidb pr title. During batch merge, multiple tidb PRs will be processed simultaneously. At this checkout step, we need to obtain all tidb pr titles to determine if the tidb-test PR has been specified.

Which situation is not allowed?
during batch merge, it is not allowed to specify the tidb-test branch or commit in the pr title for checking out code.

### CODE REVIEW

**Comment by purelind** [comment]
test on batch merge (one tidb PR title specify tidb-test PR ) https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-7.5/job/ghpr_mysql_test/641/console
test on batch merge (two tidb PR title specify tidb-test PR)  batch merge multi tidb-test PR   https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-7.5/job/ghpr_mysql_test/648/console
test on normal (one tidb PR title specify tidb-test PR) https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-7.5/job/ghpr_mysql_test/649/console
test on normal (one tidb PR not specify tidb-test PR) https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/release-7.5/job/ghpr_mysql_test/650/console

**Comment by purelind** [comment]
/hold

need more tests and postsubmit to auto merge tidb-test PR

File: libraries/tipipeline/vars/component.groovy [hunk 1]
```diff
@@ -96,6 +96,136 @@ def checkoutV2(gitUrl, keyInComment, prTargetBranch, prCommentBody, credentialsI
...
+            ], 
+            submoduleCfg: [],
+            userRemoteConfigs: [[
+                credentialsId: credentialsId,
+                refspec: refSpec,
+                url: gitUrl,
+            ]]
+        ]
+    )
+}
```
Comment (wuhuizuo) [inline libraries/tipipeline/vars/component.groovy:163]: It's better to unify the usage for git cloning and checkout, please create a backlog issue task

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**